ddtrace/tracer/spancontext.go

```diff
@@ -421,8 +440,8 @@ func samplerToDM(sampler samplernames.SamplerName) string {
 	return "-" + strconv.Itoa(int(sampler))
 }
 
-func (t *trace) setSamplingPriorityLocked(p int, sampler samplernames.SamplerName) bool {
-	if t.locked {
+func (t *trace) setSamplingPriorityWithForce(p int, sampler samplernames.SamplerName, force bool) bool {
```

Comment:
Can we add a comment above this function explaining when to use it and when NOT to use it?

At this point, we have **so many** methods for applying sampling prio, at **so many** different levels. So it's important to document what each should be used for, and when